### DISCUSSION TITLE
데이터의 저장과 처리 관련 어려움

### DISCUSSION THREAD

**Discussion by kyalivia** [devrel-kr/HackaLearn#149] [7 Q&A - 백엔드] [answered]
안녕하세요.

CRUD기반의 게시판을 만드는 중입니다. azure function을 사용해서 데이터를 처리하려고 하는데, 새로운 데이터 혹은 수정된 데이터를 저장하는 부분에서 어떻게 해결을 해야할 지 잘 모르겠습니다. azure function을 이용하여 새로운 데이터를 만들거나 수정하면, 30분 후에 기본 데이터로 돌아가고 있습니다. 수정된 데이터의 저장과, 새롭게 만든 데이터의 저장 관련하여 조언 부탁드립니다.

**Comment by justinyoo** [comment]
30분 후에 기본 데이터로 돌아간다는 말이 이해가 가지 않습니다. 좀 더 구체적으로 질문을 남겨주시거나 코드 리포지토리를 보여주시면 알 수 있을 것 같아요.

**Comment by minhoryang** [comment, marked as answer]
**별도의 sql/nosql server를 사용하셔야합니다.**

Static Webapp의 기본 동작방식은 serverless입니다. Static build(가 된 Frontend의 경우)가 아닌 API로 띄운다고 해도("전 API서버를 띄운게 아닌가요?"), 해당 인스턴스는 지속되지 않습니다.

영속되는 서버가 존재하는 방식이 아니며, 서버가 그 순간에 존재하는 척 하는 방식이기에, API를 통해 서버의 어디론가 (메모리, 변수, 파일 등)에 저장을 하였다면, 해당 서버가 할당 해제되었을 때 (timeout 등) 또는 서비스에 요청이 폭증하여 새 서버를 할당할 때에는 저장해놓은 값에 접근할 수 없게 됩니다. (값이 저장된 서버 vs 방금 막 띄워진 서버, 동시에 존재하기에 사용자는 둘 중 어디에 접속하게될지 모름)

**별도의 sql/nosql server를 사용하셔야합니다.**

**Reply by minhoryang** [reply]
- Azure SQL Database
- Azure Database for MySQL/MariaDB
- Azure Database for PostgreSQL
- Azure Cosmos DB (NoSQL, like MongoDB, Graph)
- Azure Cache for Redis (NoSQL, Redis)
- Azure Blob Storage (Object Storage, like AWS S3)

https://microsoft.github.io/AzureTipsAndTricks/blog/tip295.html

**Reply by minhoryang** [reply]
Serverless라는 개념에 대해 쉽게 설명을 하려하였지만, 혹시나 이해가 안되신다면 몇가지 소개링크를 올려드리겠습니다. 여전히 문제가 해결이 안된다면 또 물어봐주세요.

https://serverless-stack.com/chapters/ko/what-is-serverless.html

https://serverless-stack.com/chapters/ko/why-create-serverless-apps.html

https://blog.aliencube.org/ko/2016/06/23/serverless-architectures/

**Reply by kyalivia** [reply]
네, 알겠습니다.

**Reply by justinyoo** [reply]
맨 마지막의 링크는 아주 좋은 링크로군요? 🙈

**Reply by kooljay82** [reply]
위의 링크들은 다른 분들에게도 많은 도움이 될것 같아요!!! 좋은 자료입니다!!!

**Reply by jung-youjin** [reply]
좋은 링크 공유 감사합니다 😎💪🏻
